pht_federated/aggregator/clients/discovery_client.py

```diff
@@ -0,0 +1,97 @@
+import requests
+import os
+from typing import Union, List
+from uuid import uuid4
+from requests.models import Response
+
+class DiscoveryClient:
+
+    def __init__(self, api_url: str = None, username: str = None, password: str = None):
+        """
+            Setup and verify connection parameters either based on arguments or .env vars
+        """
+
+        self.api_url = api_url if api_url else os.getenv("AGGREGATOR_API_URL")
+        if not self.api_url.startswith("http://"):
+            self.api_url = "http://" + self.api_url
+        if not self.api_url.endswith("/api/proposal"):
+            self.api_url = self.api_url + "/api/proposal"
+
+        print("SELF API URL : {}".format(self.api_url))
+
+    def post_proposal(self, proposal_id: uuid4 = None) -> Response:
+        """
+        Sending POST request to create a proposal entry in the database with defined proposal_id
+        :param proposal_id: uuid4 value that identifies proposal
+        :return: result body of POST request
+        """
+        if not proposal_id:
+            proposal_id = int(os.getenv("PROPOSAL_ID"))
+        assert proposal_id
```

Comment:
proposal id should not be an environment variable. environment variables are for static configuration